File: OOP/Seminars/HomeWork/Seminar_7/PY/ClientServiceImpl.py
from ClientService import *
from ClientData import *
from Client import *


class ClientServiceImpl(ClientService):
    clientData = ClientData

    def add_client(self, Client):
            self.clientData.add_client(Client)


    def remove_client(self, client):
        if client != None:
            self.clientData.remove_client(client)
        else:
            print("Some error with removing client. ---client = null---")

    def show_clients(self):
        self.clientData.show_clients()

    def get_clients(self):
        return self.clientData.get_clients()

    def client_change_balance(self, id, value):
        self.clientData.client_change_balance(id, value)

File: OOP/Seminars/HomeWork/Seminar_7/Java/service/ClientServiceImpl.java
package OOP.Seminars.HomeWork.Seminar_7.Java.service;

import OOP.Seminars.HomeWork.Seminar_7.Java.dataBase.ClientData;
import OOP.Seminars.HomeWork.Seminar_7.Java.model.Client;

import java.util.List;

public class ClientServiceImpl implements ClientService{
    ClientData clientData = new ClientData();
    @Override
    public void addClient(Client client) {
        if(client != null){
        clientData.addClient(client);}
        else {
            System.out.println("Some error with adding client. ---client = null---");
        }
    }

    @Override
    public void removeClient(Client client) {
        if(client != null){
            clientData.removeClient(client);}
        else {
            System.out.println("Some error with removing client. ---client = null---");
        }
    }

    @Override
    public void showClients() {
        clientData.showClients();
    }

    @Override
    public List<Client> getClients() {
        return clientData.getClients();
    }

    @Override
    public void clientChangeBalance(int id, int value) {
        clientData.clientChangeBalance(id, value);
    }
    }